mlir/test/CMakeLists.txt

```diff
@@ -28,6 +27,16 @@ set(MLIR_ROCM_WRAPPER_LIBRARY_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 set(MLIR_SPIRV_WRAPPER_LIBRARY_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 set(MLIR_VULKAN_WRAPPER_LIBRARY_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 
+# Passed to lit.site.cfg.py.in to set the -x2 flag for the mlir E2E tests.
+if(MLIR_MIOPEN_DRIVER_XDLOPS_TEST_ENABLED)
+  set(MLIR_XDLOPS "-x2")
+endif()
+
+# Passed to lit.site.cfg.py.in to set the -rand flag for the mlir E2E tests.
+if(NOT MLIR_MIOPEN_DRIVER_RANDOM_DATA_SEED STREQUAL "none")
+  set(MLIR_RANDOM_DATA "-rand ${MLIR_MIOPEN_DRIVER_RANDOM_DATA_SEED} ")
```

Comment:
In our builds, this deterministically sets the random number generator to the seed `1`. Looking at the documentation, we should be passing "0" or "none" to get a deterministic 0xfffffff... seed, or passing no argument at all to get actual randomness. Do we want the random builds to actually be random?

Reply:
(Reading of the driver code has led to the conclusion that the documentation and the code there are somewhat out of sync, I'll debug that)

Reply:
-rand none or no argument at all, mlir-miopen-driver will use fixed data (not use RNG). -rand 0 or positive number will set the seed to that number. Otherwise will set the seed to time(0)

Reply:
@yiqian1 you should put a link to the comment in `mlir-miopen-driver` here

Reply:
Yep - but the code seems to disagree with this.

Reply:
Probably the wording in mlir-miopen-driver is misleading: If not specified or -rand none, use all ones as input data, not use all ones as the seed of random number generator.

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp#L265

Reply:
https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp#L1315 is concerning me

Reply:
True randomness isn't wanted right now, closing